docs/index.md

```diff
@@ -29,7 +29,46 @@ resource "illumio-cloudsecure_aws_account" "example" {
 }
 ```
 
+## Authentication and Configuration
+
+Illumio CloudSecure provides several methods for configuring the Terraform provider:
+
+1. Parameters in the provider configuration
+2. Environment variables
+
+Illumio CloudSecure follows the industry-standard protocol for authorization using OAuth 2.0. To generate the necessary client_id and client_secret to use the CloudSecure Terraform provider, you will need to create a Service Account from the [console](https://console.illum.io/#/serviceAccounts) and generate a new secret.
+
+| :warning: WARNING:                                                                                                                                                        |
+| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Hard-coded credentials are not recommended in any Terraform configuration and risks secret leakage should this file ever be committed to a public version control system. |
+
+Credentials can be provided by adding a client_id, client_secret, and optionally token, to the illumio-cloudsecure provider block.
+
+Usage:
+
+```terraform
+provider "illumio-cloudsecure" {
+  client_id = "my-access-id"
+  client_secret = "my-secret-id"
+}
+```
+
+### Environment Variables
+
+Credentials can be provided by using the `ILLUMIO_CLOUDSECURE_CLIENT_ID` and `ILLUMIO_CLOUDSECURE_CLIENT_SECRET` environment variables.
+
+```terraform
+provider "illumio-cloudsecure" {}
+```
+
+```
+% export ILLUMIO_CLOUDSECURE_CLIENT_ID="my-access-id"
+% export ILLUMIO_CLOUDSECURE_CLIENT_SECRET="my-secret-id"
+% terraform plan
+```
+
```

Comment:
Also document the option to directly pass the access token instead of `client_id`+`client_secret`.
Note that this requires the user to call the OAuth 2 token endpoint on their own at `https://cloud.illum.io/api/v1/authenticate` with the `client_id` and `client_secret` to get the access token.

```
variable "illumio_cloudsecure_access_token" {
  type        = string
  sensitive   = true
  description = "The OAuth 2 access token used to authenticate against the CloudSecure Config API."
}

provider "illumio-cloudsecure" {
  access_token = $illumio_cloudsecure_access_token
}
```

```
% export TF_VAR_illumio_cloudsecure_access_token="my-access-token"
% terraform plan
```

Reply:
BTW I realized that I need to fix the default token endpoint in the code:
https://github.com/illumio/terraform-provider-illumio-cloudsecure/blob/d1a93d181d1ec086008c8f1583fe2f9321684da1/internal/provider/provider.go#L83-L84
I will send a PR to fix that.